lib/annotate_rb/options.rb

```diff
@@ -92,102 +94,61 @@ def from(options = {}, state = {})
 
     DEFAULT_OPTIONS = {}.merge(POSITION_OPTIONS, FLAG_OPTIONS, OTHER_OPTIONS, PATH_OPTIONS).freeze
 
-    FLAG_OPTION_KEYS = [
-      :classified_sort,
-      :exclude_controllers,
-      :exclude_factories,
-      :exclude_fixtures,
-      :exclude_helpers,
-      :exclude_scaffolds,
-      :exclude_serializers,
-      :exclude_sti_subclasses,
-      :exclude_tests,
-      :force,
-      :format_markdown,
-      :format_rdoc,
-      :format_yard,
-      :frozen,
-      :ignore_model_sub_dir,
-      :ignore_unknown_models,
-      :include_version,
-      :show_check_constraints,
-      :show_complete_foreign_keys,
-      :show_foreign_keys,
-      :show_indexes,
-      :simple_indexes,
-      :sort,
-      :timestamp,
-      :trace,
-      :with_comment,
-      :with_column_comments,
-      :with_table_comments
-    ].freeze
-
-    OTHER_OPTION_KEYS = [
-      :active_admin,
-      :command,
-      :debug,
-      :hide_default_column_types,
-      :hide_limit_column_types,
-      :ignore_columns,
-      :ignore_routes,
-      :ignore_unknown_models,
-      :models,
-      :routes,
-      :skip_on_db_migrate,
-      :target_action,
-      :wrapper,
-      :wrapper_close,
-      :wrapper_open,
-      :classes_default_to_s
-    ].freeze
-
-    PATH_OPTION_KEYS = [
-      :additional_file_patterns,
-      :model_dir,
-      :require,
-      :root_dir
-    ].freeze
-
-    ALL_OPTION_KEYS = [
-      POSITION_OPTIONS.keys, FLAG_OPTION_KEYS, OTHER_OPTION_KEYS, PATH_OPTION_KEYS
-    ].flatten.freeze
-
     POSITION_DEFAULT = "before"
 
     # Want this to be read only after initializing
-    def_delegator :@options, :[]
+    def_delegators :@options, :[], :to_h
```

Comment:
I'd definitely like your thoughts here -- 

So when I refactored the old gem, I intentionally made `Options` read only. In the old gem, it was not clear where options were being set as it was a read-writeable global variable, which made it really hard to debug.

However, we are no longer in the old gem and so it might be less of a concern. Thoughts on keeping it the same versus making it writable by delegating `to_h`?

Reply:
That's a good question. Off the top of my head, I don't believe I wanted this to be writable but a convenience to get a copy. I can't see where I've even used it in the PR, so it might have been something I failed to cleanup after debugging in the console. I think your initial intention is good, I can't think of a reason `Options` should need to change after they've been initialized.